src/circadian_model.R

```diff
@@ -51,41 +91,116 @@ circadianModel <- function(t, x, params){
     R1tot <- x[5]
 
     # Get R1b from R1tot and A
-    # TODO: R1b <- ...
+    R1b <- R1b <- 0.5 * (A + R1tot + 4 - sqrt(pow(A + R1tot + 4,2.0) - 4 * A * R1tot))
 
     # Determine if student is in school hours and if it is a school day
-    # TODO: inSchoolHours <- ...
-    # TODO: isSchoolDay <- ...
+    inSchoolHours <- (t %% 24 > schoolStartLocalTimeInHours) && (t %% 24 <= schoolStartLocalTimeInHours + schoolDurationInHours)
+    isSchoolDay <- ((t %/% 24) %% 7) < 5
 
     # Calculate the light intensity
-    # TODO: light <- ...
+    light <- lightApprox(t %% 24)$y
 
     # Determine if student is awake
-    # TODO: isAwake <- ...
+    if (inSchoolHours && isASchoolDay) {
+      isAwake <<- TRUE
+      light <- schoolBrightnessInLux
+    } else {
+      
+      # If it's free time, and you want to fall asleep, you can, as long as you're not up for social reasons
+      if (sleepDrive(R1b, Psi) > Dsleep) {  
+        if (t %% 24 > socialFactor || t %% 24 < 10) { # Forbid falling asleep before a certain time due to social reasons
+          isAwake <<- FALSE
+        }
+      }
+      # If it's free time, and you want to wake up, you can wake up
+      if (sleepDrive(R1b, Psi) < Dwake) {   
+        isAwake <<- TRUE
+      }
+    }
 
     # Set mu and ichi parameters based on whether student is awake or not
-    # TODO: mu <- ...
-    # TODO: ichi <- ...
+     
+    if (isAwake) {
+      ichi <- 1.0/18.18
```

Comment:
We should define four parameters called `chiAwake`, `muAwake`, chiAsleep`, `muAsleep` at the top and then call them here so we don't have these values hardcoded in the middle of the code.

Reply:
Oh, ok. Yeah, I'll assign their values at the top, as globals, and then referencing them should be easier for these parts of the code.